src/compiler/transformers/utilities.ts

```diff
@@ -364,4 +384,24 @@ namespace ts {
         return member.kind === SyntaxKind.PropertyDeclaration
             && (<PropertyDeclaration>member).initializer !== undefined;
     }
+
+    /**
+     * Gets a value indicating whether a class element is either a static or an instance method declaration.
+     *
+     * @param member The class element node.
+     * @param isStatic A value indicating whether the member should be a static or instance member.
+     */
+    export function isStaticMethodDeclaration(member: ClassElement, isStatic: boolean): member is MethodDeclaration {

```

Comment:
The name of this function doesn't exactly match the comment. Also, the type assertion means that a valid method declaration that doesn't match the `static` flag would not be considered a `MethodDeclaration`.

You might be better off with two functions, and two "branded" versions of `MethodDeclaration`, i.e.:

```ts
export function isStaticMethodDeclaration(member: ClassElement): member is StaticMethodDeclaration { ... }
export function isNonStaticMethodDeclaration(member: ClassElement): member is NonStaticMethodDeclaration { ... }

// types.ts
/* @internal */
export interface StaticMethodDeclaration extends MethodDeclaration {
  _staticBrand: true;
}
/* @internal */
export interface NonStaticMethodDeclaration extends MethodDeclaration {
  _staticBrand: false; 
}
```

The upside of the single argument functions is that you can pass them to `filter` without a closure, i.e.:
```ts
filter(node.members, isNonStaticMethodDeclaration)
```

Alternatively, since these are only used from one code path, you might want to just consolidate them as I proposed in another comment in classFields.ts.